common/data/userDetailsCache.js

```diff
@@ -0,0 +1,33 @@
+const redis = require('./redis')
+const User = require('../models/user')
+const { REFRESH_TOKEN_LIFETIME_SECONDS } = require('../utils/constants')
+
+const userDetailsCache = async (userId, oasysUser) => {
+  const userDetails = {
+    isActive: oasysUser?.accountStatus === 'ACTIVE',
+    email: oasysUser?.email,
+    oasysUserCode: oasysUser?.oasysUserCode,
+    username: `${oasysUser?.userForename1} ${oasysUser?.userFamilyName}`,
+  }
+
+  await redis.set(`user:${userId}`, JSON.stringify(userDetails), 'EX', REFRESH_TOKEN_LIFETIME_SECONDS)
+
+  return userDetails
+}
+
+const cacheUserDetailsWithRegion = async (userId, areaCode, areaName) => {
+  const serializedDetails = await redis.get(`user:${userId}`)
+  const userDetails = new User().withDetails(JSON.parse(serializedDetails)).setArea({ areaCode, areaName })
```

Comment:
Just looking at this.. I'm wondering if `userDetails` contains the entire user object, I would expect to see to see a `getDetails` call to prevent putting tokens etc.. in the user details cache. I'll check with Redis locally

Reply:
We don't set the tokens in but this is something i was thinking about, we should potentially separate the user-auth context from the userDetails and then only combine them if we need to for the session. But keep them like 2 different objects to avoid as you said putting unnecessary info in the cache. Will refactor this if you are happy and we can discuss again. :)

Reply:
Just pulled the data for the keys from Redis
*get user:1*
```
"{\"isActive\":true,\"email\":\"ray.arnold@foo.bar\",\"oasysUserCode\":\"SUPPORT1\",\"username\":\"Ray Arnold\",\"areaCode\":\"LAN\",\"areaName\":\"Lancashire\"}"
```
*get sess:{sessionId}*
```
{\"user\":{\"id\":\"1\",\"token\":\"{token}\",\"refreshToken\":\"{refreshToken}\",\"tokenLifetime\":3599,\"tokenExpiryTime\":1622565765408}}
```
So looks like the cache is remaining clean, taking a second look at the usage here `new User().withDetails(details).setArea(area)` would have `undefined` for the token etc.. which get's omitted when doing the `JSON.stringify(user)`

Reply:
We can be explicit with `new User().withDetails(details).setArea(area).getDetails()`, but looks like things are working here already